### DISCUSSION TITLE
useForm vs useController which one I should pick?

### DISCUSSION THREAD

**Discussion by MikePrus** [react-hook-form/react-hook-form#6927] [Q&A] [answered]
Hi, I've been faced with an issue where I wanna create separate components with their own validation, and just use them across my app in different forms. 
For example, create separate email input component with its own validation. When I have two forms login & sign up I can just use one component on both forms. The main doubts about which hook fits better for my case useForm or useController. Also, I will be appreciated if somebody could provide some simple examples in codeSandBox.

**Comment by bluebill1049** [comment, marked as answer]
`useForm` is the primary hook, `useContoller` is only made for controlled components.

**Reply by MikePrus** [reply]
Do you have some examples?

**Reply by ryanmk54** [reply]
useController can be used in two separate forms. You just need to pass the control object returned from useForm.

**Reply by bluebill1049** [reply]
refer to the doc

https://react-hook-form.com/api/useform (per form)
https://react-hook-form.com/api/usecontroller (per controlled field/input)

**Reply by MikePrus** [reply]
Thanks, @ryanmk54 and @bluebill1049  for a quick answer.
